lib/urlwatch/reporters.py

```diff
@@ -108,6 +108,10 @@ def convert(self, othercls):
 
         return othercls(self.report, config, self.job_states, self.duration)
 
+    @classmethod
+    def get_base_config(cls, report):
+        return report.config['report'][cls.mro()[-3].__kind__]
```

Comment:
Will this break when someone adds another layer of subclassing? Would `cls.mro()[2]` (or 1 or whatever the number is that selects the "direct" subclass of `ReporterBase`) be better? Or should we just pick the first one that is set?

Or something like this (untested):

```
next(superclass.__kind__ for superclass in cls.mro() if hasattr(superclass, '__kind__'))
```

Reply:
I believe -3 *is* "the number is that selects the "direct" subclass of `ReporterBase`".

- -1 -> `object`
- -2 -> `ReporterBase`
- -3 -> "base kind"

```py
>>> from urlwatch.reporters import ShellReporter, TextReporter
>>> ShellReporter.mro()
[<class 'urlwatch.reporters.ShellReporter'>,
 <class 'urlwatch.reporters.TextReporter'>,
 <class 'urlwatch.reporters.ReporterBase'>,
 <class 'object'>]
>>> ShellReporter.mro()[-3]
<class 'urlwatch.reporters.TextReporter'>
>>> TextReporter.mro()
[<class 'urlwatch.reporters.TextReporter'>,
 <class 'urlwatch.reporters.ReporterBase'>,
 <class 'object'>]
>>> TextReporter.mro()[-3]
<class 'urlwatch.reporters.TextReporter'>
```

Reply:
Ah yeah, makes sense that it is in this order. Then it's all good :)